core-bundle/contao/forms/Form.php

```diff
@@ -325,10 +325,7 @@ protected function processFormData($arrSubmitted, $arrLabels, $arrFields, $arrFi
 
 		// Store submitted data (possibly modified by hook or data added) in the session for 10 seconds,
 		// so it can be used on any forward page using the {{form_session_data::<form-field-name>}} insert tag
-		if ($request = System::getContainer()->get('request_stack')->getCurrentRequest())
-		{
-			$request->getSession()->set(self::SESSION_KEY, new AutoExpiringAttribute(10, $arrSubmitted));
-		}
+		System::getContainer()->get('request_stack')->getSession()->set(self::SESSION_KEY, new AutoExpiringAttribute(10, $arrSubmitted));
```

Comment:
Would this now fail if there is no request in the stack?

Reply:
Yes, but we use the same code in multiple other places as well. Sometimes we check if the request stack contained a request and sometimes we don‘t. Pretty inconsistent. 🙈 

In the `Form` class, there will always be a request, therefore we don‘t have to check. I can revert the change if you feel better, though.

Reply:
> In the Form class, there will always be a request, therefore we don‘t have to check. But I can revert the change if you feel better then.

Not if the Form is inserted as a content element within an event and you have a RSS feed for events with detailed content.

Reply:
At this very place in the code, there will always be a request because `processFormData()` will only be called if `Input::post('FORM_SUBMIT') == $formId`!

Reply:
Agree 👍
Was just curious because of the check for the request in the current version.